src/components/VegaGraph.tsx

```diff
@@ -0,0 +1,448 @@
+        {"category": "F", "amount": 53},
+        {"category": "G", "amount": 19}
+      ];
+      setBarData(JSON.stringify(dataArray, undefined, 1));
+      dataArray = JSON.stringify(dataArray, undefined, 1)
+    }else{
+      dataArray = barData;
+    }
+    setGraphDataArray(dataArray);
+
+    specs = {
+      "$schema": "https://vega.github.io/schema/vega/v5.json",
+      "description": "A basic bar chart example, with value labels shown upon mouse hover.",
+      "width": 400,
+      "height": 400,
+      "padding": 5,
+      "data": [
+        {
+          "name": "table",
+          "values": dataArray
+        }
+      ],    
+      "signals": [
+        {
+          "name": "tooltip",
+          "value": {},
+          "on": [
+            {"events": "rect:mouseover", "update": "datum"},
+            {"events": "rect:mouseout",  "update": "{}"}
+          ]
+        }
+      ],    
+      "scales": [
+        {
+          "name": "xscale",
+          "type": "band",
+          "domain": {"data": "table", "field": "category"},
+          "range": "width",
+          "padding": 0.05,
+          "round": true
+        },
+        {
+          "name": "yscale",
+          "domain": {"data": "table", "field": "amount"},
+          "nice": true,
+          "range": "height"
+        }
+      ],    
+      "axes": [
+        { "orient": "bottom", "scale": "xscale" },
+        { "orient": "left", "scale": "yscale" }
+      ],    
+      "marks": [
+        {
+          "type": "rect",
+          "from": {"data":"table"},
+          "encode": {
+            "enter": {
+              "x": {"scale": "xscale", "field": "category"},
+              "width": {"scale": "xscale", "band": 1},
+              "y": {"scale": "yscale", "field": "amount"},
+              "y2": {"scale": "yscale", "value": 0}
+            },
+            "update": {
+              "fill": {"value": "steelblue"}
+            },
+            "hover": {
+              "fill": {"value": "red"}
+            }
+          }
+        },
+        {
+          "type": "text",
+          "encode": {
+            "enter": {
+              "align": {"value": "center"},
+              "baseline": {"value": "bottom"},
+              "fill": {"value": "#333"}
+            },
+            "update": {
+              "x": {"scale": "xscale", "signal": "tooltip.category", "band": 0.5},
+              "y": {"scale": "yscale", "signal": "tooltip.amount", "offset": -2},
+              "text": {"signal": "tooltip.amount"},
+              "fillOpacity": [
+                {"test": "datum === tooltip", "value": 0},
+                {"value": 1}
+              ]
+            }
+          }
+        }
+      ]
+    }    
+    const runtime = parse(specs);    
+    const graphVega = document.getElementById('graphVega');
+    var viewGraph = new View(runtime)
+                .renderer('svg')     // set render type (defaults to 'canvas')
+                .initialize(graphVega);
+    viewGraph.run();
```

Comment:
[I found a component for this that could work better! ](https://github.com/vega/react-vega) Let's use it instead of trying to redo the wheel.